### DISCUSSION TITLE
Example code for using Planetscale with sqlalchemy

### DISCUSSION THREAD

**Discussion by morrisclay** [planetscale/discussion#259] [Q&A] [answered]
Hi Community,

I am looking for some example code for using Planetscale with sqlalchemy which explains how to work with the ssl restrictions. 

My current code...
```python
args = {
            'ssl_mode':'VERIFY_IDENTITY', 
            'ssl':{'ca': '/etc/ssl/cert.pem'}
        }

engine = create_engine(f'mysql+pymysql://{user}:{pass}@{host}/{db}?ssl=true', )
```
produces the error below...

```
File "/.local/lib/python3.7/site-packages/pymysql/connections.py", line 365, in _create_ssl_ctx
    ca = sslp.get("ca")
AttributeError: 'str' object has no attribute 'get'
```

Thanks for your help.
Mocl

**Comment by jgreet** [comment, marked as answer]
Are you passing in your connection arguments? It doesn't look like PyMySQL will [accept `ssl_mode`](https://github.com/PyMySQL/PyMySQL/blob/dd47caae95011e79b9e2ee12549d23f05a7f839d/pymysql/connections.py#L129-L135). 

I'm not very familiar with SQLAlchemy but copying an example from [the docs](https://docs.sqlalchemy.org/en/14/core/connections.html) this seems to connect OK.

```python
from sqlalchemy import create_engine
from sqlalchemy import text


args = {
            'ssl': {'ca': '/etc/ssl/cert.pem'}
        }

engine = create_engine('mysql+pymysql://{user}:{pass}@{host}/{db}', connect_args=args)

with engine.connect() as connection:
    result = connection.execute(text('SELECT @@version'))
    for row in result:
      print(row)
```

**Reply by morrisclay** [reply]
thanks. worked for me!
